exporter/elasticsearchexporter/exporter.go

```diff
@@ -36,16 +55,41 @@ type elasticsearchExporter struct {
 	allowedMappingModes map[string]MappingMode
 	bulkIndexers        bulkIndexers
 	bufferPool          *pool.BufferPool
+
+	// Data cache for profiles
+	knownTraces      *lru.LRUSet[string]
+	knownFrames      *lru.LRUSet[string]
+	knownExecutables *lru.LRUSet[string]
 }
 
 func newExporter(
 	cfg *Config,
 	set exporter.Settings,
 	index string,
 	dynamicIndex bool,
-) *elasticsearchExporter {
+) (*elasticsearchExporter, error) {
 	allowedMappingModes := cfg.allowedMappingModes()
 	defaultMappingMode := allowedMappingModes[canonicalMappingModeName(cfg.Mapping.Mode)]
+
+	// Create LRUs with MinILMRolloverTime as lifetime to avoid losing data by ILM roll-over.
+	knownTraces, err := freelru.New[string, lru.Void](knownTracesCacheSize, stringHashFn)
```

Comment:
Does it make sense to make LRUSet a wrapper around freeLRU, so that Void doesn't need to be exposed here?